source/adios2/toolkit/format/bp/bp4/BP4Deserializer.tcc

```diff
@@ -98,18 +98,14 @@ BP4Deserializer::InitVariableBlockInfo(core::Variable<T> &variable,
         const std::vector<typename core::Variable<T>::Info> blocksInfo =
             BlocksInfo(variable, stepsStart);
 
-        if (m_DebugMode)
+        if (variable.m_BlockID >= blocksInfo.size())
```

Comment:
I'd rather control all user input check with a single bool. This breaks this internal pattern.